objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/ObjectRecognitionCore.java

```diff
@@ -390,6 +401,302 @@ public void sendDataFromImage(ImageRetrieverResults imageResults, ExtensionServi
+                       + "[<yourStartDate>, <yourEndDate>].", null);
+               return null;
+           }
+       } else {
+           client.sendQueryError(replyAddress, "io.vantiq.extsrc.objectRecognition.invalidQueryRequest", 
+                   "No imageName or imageDate was specified, or they were incorrectly specified. "
+                   + "Cannot select image(s) to be " + operation + " .", null);
+           return null;
+       }
+       
+       Map<String, Object> parameterValues = new LinkedHashMap<String,Object>();
+       parameterValues.put("imageDir", imageDir);
+       parameterValues.put("imageName", imageName);
+       parameterValues.put("imageDate", imageDate);
+       parameterValues.put("imageUtil", imageUtil);
+       
+       return parameterValues;
+   }
+   
+   /**
+    * A helper method called by uploadLocalImages, used to process parameters and call methods that upload images
+    * @param imageDir       Directory containing the local images
+    * @param imageName      The imageName query parameter if specified, otherwise null
+    * @param imageDate      The imageDate query parameter if specified, otherwise null
+    * @param imageUtil      The instantiated ImageUtil class, setup with properties based on the request
+    * @param replyAddress   The replyAddress used to send a response to the query
+    */
+   public void processUploadQuery(String imageDir, String imageName, List<String> imageDate, ImageUtil imageUtil, String replyAddress) {
+       // Used to decide which upload helper method to call
+       boolean uploadOne = true;
+       List<Date> dateRange = new ArrayList<Date>();
+              
+       if (imageName != null) {
+           // If imageName is "all", then upload all images in the directory
+           if (imageName.equals("all")) {
+               uploadOne = false;
+               dateRange.add(null);
+               dateRange.add(null);
+               
+           // Otherwise, upload the one image matching the imageName.
+           } else {
+               uploadOne = true;
+           }
+       } else if (imageDate != null) {
+           for (String date : imageDate) {
+               if (date.equals("-")) {
+                   dateRange.add(null);
+               } else {
+                   try {
+                       dateRange.add(format.parse(date));
+                   } catch (ParseException e) {
+                       log.error("An error occurred while parsing the imageDate");
+                       client.sendQueryError(replyAddress, "io.vantiq.extsrc.objectRecognition.invalidQueryRequest", 
+                               "One of the dates in the imageDate list could not be parsed.Please be sure that both "
+                               + "dates are in the following format: yyyy-MM-dd--HH-mm-ss", null);
+                       return;
+                   }
+               }
+           }
```

Comment:
Is there some reason we cannot do all this processing in the handleQueryParameters() call?  That should return, in its map, either an imageName set OR an imageDate that would be a filter.  Then, if imageName is set, we upload/deleteOne (which may no longer be necessary);  otherwise, use the filter.  We may have to send in an "intent" if we need to parameterize the errors to say for "update" vs. "delete".

We have literally the same copied code both places (except for uploadOne vs deleteOne -- both which, if necessary, become "isSingle" or something).  text error compatible.